python/ray/data/tests/test_batch_mapper.py

```diff
@@ -152,10 +95,42 @@ def add_and_modify_udf_numpy(data: Union[np.ndarray, Dict[str, np.ndarray]]):
 
 
 @pytest.mark.parametrize(
-    "ds_with_expected_pandas_numpy_df",
+    "ds",
+    [
+        lazy_fixture("ds_pandas_single_column_format"),
+        lazy_fixture("ds_pandas_multi_column_format"),
+        lazy_fixture("ds_arrow_single_column_format"),
+        lazy_fixture("ds_arrow_single_column_tensor_format"),
+        lazy_fixture("ds_arrow_multi_column_format"),
+        lazy_fixture("ds_numpy_single_column_tensor_format"),
+        lazy_fixture("ds_numpy_list_of_ndarray_tensor_format"),
+    ],
+)
+def test_batch_mapper_batch_size(ray_start_regular_shared, ds):
```

Comment:
what happens here if 

1) Given batch_size is smaller than dataset block size 
2) Given batch_size is larger than dataset block size 
3) No batch_size given tested against multiple dataset sizes: < 4096 and > 4096 ?

Reply:
added more test coverage for this